sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/PropertiesAccessCorrectField.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class NonCompliantClass_FromRspec
+    {
+        private int x;
+        private int y;
+
+        public int X
+        {
+            get { return x; }
+            set { x = value; }
+        }
+
+        public int Y
+        {
+            get { return x; }  // Noncompliant {{Refactor this getter so that it actually refers to the field 'y'}}
+//                       ^
+            set { x = value; } // Noncompliant {{Refactor this setter so that it actually refers to the field 'y'}}
+//                ^
+        }
+    }
+
+    class NonCompliantStruct_FromRspec
+    {
+        private int x;
+        private int y;
+
+        public int X
+        {
+            get { return x; }
+            set { x = value; }
+        }
+
+        public int Y
+        {
+            get { return x; }  // Noncompliant: field 'y' is not used in the return value
+            set { x = value; } // Noncompliant: field 'y' is not updated
+        }
+    }
+
+
+    class NonCompliant_UnderscoresInNamesAndCasing
+    {
+        private int yyy;
+        private int __x__X; // test that underscores and casing in names are ignored
+        public int XX
+        {
+            get { return yyy; }   // Noncompliant {{Refactor this getter so that it actually refers to the field '__x__X'}}
+//                       ^^^
+            set { yyy = value; }  // Noncompliant {{Refactor this setter so that it actually refers to the field '__x__X'}}
+//                ^^^
+        }
+
+        public string _Y___Y_Y_
+        {
+            get { return __x__X; } // Noncompliant
+//                       ^^^^^^
+        }
+    }
```

Comment:
I am curious if we could detect the following:
```
private int x;
private int _x;
public int X { get { return _x; } }
public int _X { get { return x; } }
```

Reply:
We perhaps could if we broke the comparison into two and first looked for a case-insensitive match, and then again after we stripped the underscores. Have you seen many properties with that differ only by underscores in the name? (I hope not!)

Reply:
I rarely see bad code :)